src/snowflake/snowpark/modin/plugin/__init__.py

```diff
@@ -21,11 +21,16 @@
 # since modin may raise its own warnings/errors on the wrong pandas version
 import pandas  # isort: skip  # noqa: E402
 
-supported_pandas_version = "2.2.1"
-if pandas.__version__ != supported_pandas_version:
+supported_pandas_major_version = 2
+supported_pandas_minor_version = 2
+actual_pandas_version = version.parse(pandas.__version__)
```

Comment:
as we discussed before, shall we first check modin, then check pandas?

Reply:
I tried this approach (importing modin, catching the warning it raises, then re-raising an error), but it seems like modin will sometimes raise both an error AND a warning. I tried installing `pandas==1.5.3` and running `import modin.pandas`, and this is the error it gave:
```
>>> import modin.pandas
UserWarning: The pandas version installed (1.5.3) does not match the supported pandas version in Modin (2.2.X). This may cause undesired side effects!
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joshi/miniconda3/envs/snowpandas-modin-0.30.1/lib/python3.9/site-packages/modin/pandas/__init__.py", line 178, in <module>
    from .dataframe import DataFrame
  File "/Users/joshi/miniconda3/envs/snowpandas-modin-0.30.1/lib/python3.9/site-packages/modin/pandas/dataframe.py", line 64, in <module>
    from modin.pandas.io import from_non_pandas, from_pandas, to_pandas
  File "/Users/joshi/miniconda3/envs/snowpandas-modin-0.30.1/lib/python3.9/site-packages/modin/pandas/io.py", line 50, in <module>
    from pandas._typing import (
ImportError: cannot import name 'DtypeBackend' from 'pandas._typing' (/Users/joshi/miniconda3/envs/snowpandas-modin-0.30.1/lib/python3.9/site-packages/pandas/_typing.py)
```

This can be worked around (we can just catch `ImportError` in addition to checking the raised `UserWarning`), but I'm worried that re-raising with an error telling users to check the pandas version may not always be the correct fix. If some other dependency of modin is installed with the wrong version as well as pandas, then an `ImportError` might have a different meaning.

Any thoughts on how to handle this?

Reply:
Filed SNOW-1758773 to track